File: problems/java/09. Palindrome Number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0 || (x % 10 == 0 && x != 0)) {
            return false;
        }

        int revh = 0;
        while (x > revh) {
            revh = revh * 10 + (x % 10);
            x /= 10;
        }

        return x == revh || x == revh / 10;
    }
}

File: problems/Rust/09. Palindrome Number.rs
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        if x < 0 || (x % 10 == 0 && x != 0) {
            return false;
        }

        let mut x = x;
        let mut revh = 0;
        while x > revh {
            revh = revh * 10 + x % 10;
            x /= 10;
        }

        x == revh || x == revh / 10
    }
}